index.js

```diff
@@ -22,14 +33,19 @@ if (plugin.configs) {
       selfPlugin.configs[configName].extends = [].concat(config.extends)
         .map(extendsName => extendsName.replace(`plugin:${pluginName}/`, 'plugin:self/'));
     }
+    if (config.plugins) {
+      selfPlugin.configs[configName].plugins = [].concat(config.plugins)
+        .map(enabledPluginName => enabledPluginName.replace(pluginName, 'self'));
+    }
     if (config.rules) {
-      selfPlugin.configs[configName].rules = Object.assign({}, config.rules);
-      Object.keys(config.rules).forEach(ruleName => {
-        if (ruleName.startsWith(`${pluginName}/`)) {
-          selfPlugin.configs[configName].rules[`self${ruleName.slice(ruleName.lastIndexOf('/'))}`] = config.rules[ruleName];
-          delete selfPlugin.configs[configName].rules[ruleName];
-        }
-      });
+      selfPlugin.configs[configName].rules = createRuleset(config.rules);
+    }
+    if (config.overrides) {
+      selfPlugin.configs[configName].overrides = [].concat(config.overrides)
+        .map((override) => {
+          override.rules = createRuleset(override.rules);
```

Comment:
Should this shallow-copy the `override` object to avoid mutating the original config?

Reply:
`[].concat(config.overrides)` on line 44 deals with that shallow copy. That copied array is the one that is mutated, not the original `config.overrides` from the source config. The array that is created from the `contat` is never used anywhere else so it is safe to mutate it.

I agree that it'd be clearer to do this by creating a whole new array and that is easy with rest/spread operators but I wasn't sure what minimum version of node you wanted to support.

This logic is the same as how you already handle creating a new version of the `config.extends`  array. I felt it was best to mimic current patterns rather than do something new :)

Reply:
Agreed that the `config.overrides` array will never be mutated, but it seems like individual elements of the `config.overrides` array will be mutated by having the `rules` property on them set to `createRuleset(override.rules)`.

Reply:
Gah sorry, I realised you were right after posting my response. Just pushed a fix.